gammapy/maps/wcsnd.py

```diff
@@ -669,24 +669,47 @@ def cutout(self, position, width, mode="trim"):
         cutout : `~gammapy.maps.WcsNDMap`
             Cutout map
         """
-        width = _check_width(width)
-        idx = (0,) * len(self.geom.axes)
-        c2d = Cutout2D(
-            data=self.data[idx],
-            wcs=self.geom.wcs,
-            position=position,
-            # Cutout2D takes size with order (lat, lon)
-            size=width[::-1] * u.deg,
-            mode=mode,
-        )
+        geom_cutout = self.geom.cutout(position=position, width=width, mode=mode)
 
-        # Create the slices with the non-spatial axis
-        cutout_slices = Ellipsis, c2d.slices_original[0], c2d.slices_original[1]
+        slices = geom_cutout.cutout_info["parent-slices"]
+        cutout_slices = Ellipsis, slices[0], slices[1]
 
-        geom = WcsGeom(c2d.wcs, c2d.shape[::-1], axes=self.geom.axes)
         data = self.data[cutout_slices]
 
-        return self._init_copy(geom=geom, data=data)
+        return self._init_copy(geom=geom_cutout, data=data)
+
+    def stack(self, other, weights=None, check=True):
```

Comment:
Since this is doing the same thing as `coadd` except that it works for a cutout why not name this `coadd_cutout`? `stack` is already used for datasets. This might be confusing since this method could be used for other usages than stacking, no?

Reply:
Another option is to add a test in `coadd` so that `if self.is_cutout_from(other)` the simple slice approach is used. Then you don't need to check in each `stack` method what to use.

Reply:
I agree there is an overlap in functionality of `.stack()` and `.coadd()`. So `.coadd()` is the completely general, but slow method and `.stack()` is the method to be used in the  single use-case in combination with `.cutout()`. I have a minimal preference to not merge both methods into one for two minor reasons:
1. I like the mirrored API between `MapDataset` and `Map`, so `MapDataset.stack()`, calls into `Map.stack()` and `PSFMap.stack()` etc.
2. Adding the fast track to `.coadd()` complicates the code in `.coadd()` to a lot of cases, with checking for aligned geometries, checking if `.cutout_info` is set etc. I think it's easier to maintain the `.cutout()` and `.stack()` use-case in a separate place.